pilot/test/integration/cloudfoundry/copilot_test.go

```diff
@@ -52,7 +52,39 @@ const (
 	appEnvoyListener  = "http://127.0.0.1:6666"
 )
 
-func TestEdgeRouterWithMockCopilot(t *testing.T) {
+var routeRule = `
+apiVersion: config.istio.io/v1alpha2
+kind: V1alpha2RouteRule
+metadata:
+  name: ultimate-route-rule
+spec:
+  hosts:
+  - abc.com
+  gateways:
+  - ultimate-gateway
+  http:
+  - route:
+    - destination:
+        port:
+          number: 10655
+        name: abc.com
+`
+var gatewayConfig = `
+apiVersion: config.istio.io/v1alpha2
+kind: Gateway
+metadata:
+  name: ultimate-gateway
+spec:
+  servers:
+  - port:
+      number: 6666
+      protocol: http
+      name: "10655"
```

Comment:
@rshriram we got a bit of surprising behavior when we tried to leave the name off of this. Since there is only one service port we would have expected that it would have made its way into the route rule specified above this but instead we received a panic. I didn't have time to actually look at where the panic was coming from but will investigate more tomorrow.

Also, the other strange bit of behavior here is that the name has to be a string of an int due to the fact that we `strconv.Atoi` it later on. If you don't wrap this in quotes you end up getting a warning and then an empty set of `virtualHosts` when you go to get your `routes`.

cc @ZackButcher

Reply:
This is definitely not correct: name should be a string and never be a number. We should never call Atoi on a name.

Reply:
Why can't we enforce name required rule? And for ingress, we can enforce that people have to name their ports in the ingress spec (thats not much of an issue, since it involves just tweaking the config), or in worst case, we can generate uuid for the port name. It keeps the code much more simpler, and enforces some discipline in the configs as well. WE already have several precedences for this (requiring app: <name> in the deployment so that traces show up properly, and other such things)